python/lsst/ctrl/oods/fileLinker.py

```diff
@@ -0,0 +1,76 @@
+# This file is part of ctrl_oods
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import os
+from lsst.ctrl.oods.fileQueue import FileQueue
+
+LOGGER = logging.getLogger(__name__)
+
+
+class FileLinker():
+    def __init__(self, config):
+
+        staging_dir = config["stagingDirectory"]
+        self.link_dirs = config["linkToDirectories"]
+        file_queue = FileQueue(staging_dir)
+
+        asyncio.create_task(file_queue.queue_files())
+
+        asyncio.create_task(self._linker(file_queue.dequeue_file))
+
+    async def _linker(self, method):
+        while True:
+            filename = await method()
+            dir_name = self.create_landing_directory(filename)
+            for link_dir in self.link_dirs:
+                os.link(filename, os.path.join(link_dir, dir_name, os.path.basename(filename)))
+
+    def create_landing_directory(self, root_dir, filename):
```

Comment:
If it is a public method, it needs a docstring. If it is a private one, it should probably be prefixed with an `_`.

Reply:
I added the _

Reply:
This file will be removed, since it's not in use yet.